### PR TITLE
186.change map marker color

### LINKED ISSUE #186: Programmatically Change Map Marker Color
[//]: # (These are comments and are used for reference and will not show up in the PR)

User Requirement: 06.01.02

[//]: # (Please Add labels for type of issue)

Description: Right now we layer the default map marker underneath the emoticon for a Vibe. The default map marker is purple but it should match the color of the vibe, so need to somehow do that programmatically. There is an `if` statement somewhere in the `MapFragment` activity where changing the colour should be placed (when checking if `markerID != map_marker`).

Resources: https://medium.com/@emmaguy/dynamically-changing-svg-colours-on-android-b026a99137ad

[//]: # (If an issue is a bug or defect, please list steps to be able to reproduce bug and a screenshot of the defect i.e terminal message)

Story Points: 2

Related Issues: N/A

[//]: # (Notes:)
[//]: # (- Testing and Documentation tasks should always be assumed, but only need to be explicitly stated if there is something that ~~does not~~ need modification. For example,)
[//]: # (some Issues/Stories may not require one of Unit or Intent tests, and this should be indicated in the task list.)
[//]: # (- Code Reviews are also always assumed to be required, but an issue should only indicate in the description if a review is not required, but someone still shouldn't)
[//]: # (merge their own pull request)

### PR DESCRIPTION
**Pull request by raysarinas**
[//]: # (These are comments and are used for reference and will not show up in the PR)

Description: #186 . Update some documentation and overlay map markers to use a Canvas to show the emoticons on top of a black marker. Otherwise when picking a location, should still have the same purple color. 

[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change)

[//]: # (automatically closes given issue number)
Closes #186 

Proof of Concept
Screenshot_1574895694

[//]: # (Attach Image of feature added or fix applied and working)
[//]: # (Please provide images of tests passing)

Checklist

[//]: # (Add an x between square brackets to mark item as done)

- [x] Follows style guidelines and is **properly commented**
- [x] Code changes generate no new warnings
- [x] Documentation (UML, Wiki, etc.)

### CODE REVIEW

File: app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java [hunk 1]
```diff
@@ -181,6 +181,9 @@ private void setTheme(Vibe vibe) {
         addBtn.setBackgroundResource(vibe.getColor());
     }
 
+    /**
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java]: Nice

File: app/src/main/java/com/example/vybe/MapFragment.java [hunk 2]
```diff
@@ -168,13 +173,21 @@ public void onLowMemory() {
         mMapView.onLowMemory();
     }
 
+    /**
+     * sets the camera around a location
+     * @param latLng
+     * the coordinates the map should be centered at
+     */
     public void setToLocation(LatLng latLng) {
         Log.d(TAG, "setToLocation: Here");
```
Comment (jacobwong98) [inline app/src/main/java/com/example/vybe/MapFragment.java]: do y'all still need this?

File: app/src/main/java/com/example/vybe/MapFragment.java [hunk 3]
```diff
@@ -211,8 +233,14 @@ public void clearMap() {
...
-        Drawable background = ContextCompat.getDrawable(context, R.drawable.ic_map_marker);
-        // TODO: change marker background color
+
+        int mapMarker = markerID;
+        // TODO: clean up checking twice
+        if (mapMarker != R.drawable.ic_map_marker) {
+            mapMarker = R.drawable.ic_map_marker_vibe;
+        }
+
+        Drawable background = ContextCompat.getDrawable(context, mapMarker);
```
Comment (TheBroMoe) [inline app/src/main/java/com/example/vybe/MapFragment.java]: Please fix this before merging

File: app/src/main/java/com/example/vybe/MapFragment.java [hunk 4]
```diff
@@ -209,9 +209,9 @@ public void addMarker(LatLng latLng, @DrawableRes int drawableRes, String owner)
         BitmapDescriptor marker;
 
         if (drawableRes != R.drawable.ic_map_marker) {
-            marker = vectorToBitmap(getContext(), drawableRes);
+            marker = emoticonVectorToBitmap(getContext(), drawableRes);
         } else {
-            marker = vectorToBitmap(getContext());
+            marker = defaultVectorToBitmap(getContext());
```
Comment (TheBroMoe) [inline app/src/main/java/com/example/vybe/MapFragment.java]: Nice

**Pull request merged**